pandas/core/indexes/range.py

```diff
@@ -380,8 +380,15 @@ def get_loc(self, key, method=None, tolerance=None):
 
     @Appender(_index_shared_docs["get_indexer"])
     def get_indexer(self, target, method=None, limit=None, tolerance=None):
-        if not (method is None and tolerance is None and is_list_like(target)):
-            return super().get_indexer(target, method=method, tolerance=tolerance)
+        if not (
+            method is None
+            and tolerance is None
+            and limit is None
```

Comment:
(I think) a ``limit`` argument without a ``method`` argument is irrelevant and hence does not need to be checked.

Reply:
ahh, get it! thanks!

Reply:
It's true that `limit` without `method` is irrelevant but IMO `limit` does need to be checked here since we're deferring the validation of it to the parent class.  Note that providing `limit` without `method` will raise for other index types but will succeed for `RangeIndex` if we don't do the check here:
```python
In [2]: pd.RangeIndex(5).get_indexer([2, 3], limit=1)
Out[2]: array([2, 3])

In [3]: pd.Int64Index(range(5)).get_indexer([2, 3], limit=1)
---------------------------------------------------------------------------
ValueError: limit argument only valid if doing pad, backfill or nearest reindexing
```

Reply:
ahh, thanks for your detailed explanation, appreciate a lot!! thx!! @jschendel